tests/component/copypaste-internal.spec.tsx

```diff
@@ -0,0 +1,159 @@
+import { expect, test } from "../setup/setupScriptComponent";
+import Editor from "../utils/components/Editor";
+import { compareDocToSnapshot, focusOnEditor } from "../utils/editor";
+import { executeSlashCommand } from "../utils/slashmenu";
+import { copyPasteAll } from "../utils/copypaste";
+
+test.describe.configure({ mode: "serial" });
+
+test.beforeEach(async ({}, testInfo) => {
+  testInfo.snapshotSuffix = "";
+});
+
+test("Alert Copy/Paste Internal", async ({ browserName, mount, page }) => {
+  test.skip(
+    browserName === "firefox",
+    "Playwright doesn't correctly simulate drag events in Firefox."
+  );
+
+  await mount(<Editor blockTypes={["alert"]} />);
+
+  await focusOnEditor(page);
+  await page.keyboard.type("Paragraph 1");
+  await executeSlashCommand(page, "alert");
+  await page.keyboard.press("ArrowDown");
+  await page.keyboard.type("Alert");
+  await page.keyboard.press("ArrowDown");
+  await page.keyboard.type("Paragraph 2");
+
+  const button = await page.locator(
+    `[data-content-type="alert"] > div > div:first-child`
+  );
+  await button.first().click();
+
+  await copyPasteAll(page, "mac");
+  await page.waitForTimeout(50);
+
+  await compareDocToSnapshot(page, "alert-internal");
+});
+
+test("Button Copy/Paste Internal", async ({ browserName, mount, page }) => {
+  test.skip(
+    browserName === "firefox",
+    "Playwright doesn't correctly simulate drag events in Firefox."
+  );
+
+  await mount(<Editor blockTypes={["button"]} />);
+
+  await focusOnEditor(page);
+  await page.keyboard.type("Paragraph 1");
+  await executeSlashCommand(page, "button");
+  await page.keyboard.press("ArrowDown");
+  await page.keyboard.type("Paragraph 2");
+
+  await copyPasteAll(page, "mac");
+  await page.waitForTimeout(50);
+
+  const button = await page.locator("button");
+  await button.first().click();
+  await button.last().click();
+
+  await compareDocToSnapshot(page, "button-internal");
+});
+
+test("Embed Copy/Paste Internal", async ({ browserName, mount, page }) => {
```

Comment:
Nope, we've only really used screenshot tests before to make sure styling props and marks worked properly since that's the main thing they're for. Here imo we mostly care about the underlying TipTap/PM doc structure.

Reply:
I suppose to check rendering we could add screenshot tests but again imo it's not necessary